src/esp/assets/GenericInstanceMeshData.cpp

```diff
@@ -94,33 +105,184 @@ Cr::Containers::Optional<InstancePlyData> parsePly(
   Cr::Utility::copy(meshColors, Cr::Containers::arrayCast<Mn::Color3ub>(
                                     Cr::Containers::arrayView(data.cpu_cbo)));
 
-  /* Check we actually have object IDs before copying them, and that those are
-     in a range we expect them to be */
+  // Check we actually have object IDs before copying them, and that those are
+  // in a range we expect them to be
   data.objectIds.resize(meshData->vertexCount());
   Cr::Containers::Array<Mn::UnsignedInt> objectIds;
+
   if (meshData->hasAttribute(Mn::Trade::MeshAttribute::ObjectId)) {
     objectIds = meshData->objectIdsAsArray();
     data.objIdsFromPly = true;
+    data.objPartitionsFromSSD = false;
     const int maxVal = Mn::Math::max(objectIds);
     ESP_CHECK(
         maxVal <= 65535,
         Cr::Utility::formatString(
             "{}Object IDs can't be stored into 16 bits : Max ID Value : {}",
             msgPrefix, maxVal));
+    colorMapToUse.resize(Mn::DebugTools::ColorMap::turbo().size());
```

Comment:
Just an FYI, resize also default initializes the values in the std::vector, which seems wasteful is you are about to overwrite it. What you are looking for might be `reserve()` with `shrink_to_fit`